zellij-server/src/panes/grid.rs

```diff
@@ -1493,6 +1824,249 @@ impl Grid {
         self.scrollback_buffer_lines =
             subtract_isize_from_usize(self.scrollback_buffer_lines, transferred_rows_count);
     }
+
+    pub fn search_forward(&mut self) {
+        let search_viewport_for_the_first_time =
+            self.search_results.active.is_none() && !self.search_results.selections.is_empty();
+        let search_viewport_again = !self.search_results.selections.is_empty()
+            && self.search_results.active.is_some()
+            && self.search_results.active.as_ref() != self.search_results.selections.last();
+
+        if search_viewport_for_the_first_time || search_viewport_again {
```

Comment:
I am not 100% sure about that.
I was afraid it would be too jumpy/chaotic, since I start searching with every typed character (in `EnterSearch`-mode).

But a possibility would be to only jump to the first occurrence, once I hit Return (when leaving `EnterSearch`-mode and going to `Search`-mode)? There, I could simply call `search_backward()`, if nothing was found yet. 

What do you think?

Reply:
That can work, but just from testing in my firefox - it behaves by jumping immediately to the first occurrence if it's not on screen. I find that UX to be pretty good personally... but I can also go with the `Enter` way, as you wish.

Reply:
Can we mitigate this by stopping the search as soon as the needle is not found?

Reply:
Hm, I would need to remember the last needle that did not match anything in the scrollbuffer and if the new needle starts with the old one, return right away or something like that.
Not sure if its needed, though. There is no difference in the UX, I would say. Nothing happens either way :-) We 'only' safe some CPU cycles. 
Although, I currently don't know what happens when the user types faster than the search takes. Either he can't type while it's searching (because the UI is blocked), which would then be not a problem, or the already typed characters would trickle in one by one, causing a full search for each. That would be annoying.